### PR TITLE
week3-server-comments-box

### PR DESCRIPTION
**Pull request by josh8551021**
Added comments box to portfolio page.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (mkmg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: Remove blank line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (mkmg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Consider simply calling request.getParameter above instead of creating a helper function to do it. 

This post (see top answer in particular) has a good discussion about when to make functions containing only 1 line, and when to not:
https://softwareengineering.stackexchange.com/questions/107669/one-line-functions-that-are-called-only-once

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (mkmg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Use the List interface instead of the implementation type ArrayList. This way you can change the implementation (e.g. switch to a LinkedList) without having to change as much code.

Comment (tzsebe) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Use interface (sorry if this is a redundant comment - I think this might be one of those changes where your previous change is included).

**Review by mkmg — APPROVED** [review]
Looking good. Please address comments before submitting (and wait for Tamas to sign-off).

**Review by tzsebe — CHANGES_REQUESTED** [review]

**Comment by josh8551021** [comment]
Whoops, looks like I accidentally stayed on the same branch when I continued working. When I created a pull request, it failed, but IntelliJ still pushed the changes.

Is it alright if you guys just review the changes in the latest commit as well? Or is there a good way to reverse the pushing process?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (josh8551021) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (josh8551021) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Yeah, I had plans to expand the function to do more things (in which case, it would have been void instead), but I realized that that might not happen. You'll probably see in the changes for the next part that I accidentally pushed, I changed that.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (josh8551021) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done.

Comment (josh8551021) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done.

Comment (mkmg) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I'm not seeing this change. Did you make it on a different branch?

**Comment by mkmg** [comment]
We can just review the latest commit in this pull request.

I'm looking at the code, and I'm not seeing the change you said you made. Did you forget to add/push them?

**Comment by josh8551021** [comment]
Yeah, sorry. I forgot to commit and push my changes.

**Review by tzsebe — APPROVED** [review]

**Comment by josh8551021** [comment]
Hey, Megan. Just wanted to make sure that you looked over the changes made in commit [7823243](https://github.com/josh8551021/step/pull/4/commits/78232438a0c280a0593c456e094cfe532e912752) before merging.

**Comment by mkmg** [comment]
Thanks for checking Josh. Changes look great :)

**Pull request merged**
